module/x/gravity/types/key.go

```diff
@@ -58,108 +58,190 @@ const (
-	EthereumHeightVoteKey
+	// EVMHeightVoteKey indexes the latest heights observed by each validator
+	EVMHeightVoteKey
 )
 
 ////////////////////
 // Key Delegation //
 ////////////////////
 
+// Uint32ToBigEndian - marshals uint64 to a bigendian byte slice so it can be sorted
+func Uint32ToBigEndian(i uint32) []byte {
+	b := make([]byte, 8)
+	binary.BigEndian.PutUint32(b, i)
+	return b
+}
+
+// BigEndianToUint32 returns a uint32 from big endian encoded bytes. If encoding
+// is empty, zero is returned.
+func BigEndianToUint32(bz []byte) uint32 {
+	if len(bz) == 0 {
+		return 0
+	}
+
+	return binary.BigEndian.Uint32(bz)
+}
+
 // MakeOrchestratorValidatorAddressKey returns the following key format
 // prefix
-// [0xe8][cosmos1ahx7f8wyertuus9r20284ej0asrs085case3kn]
+// [0xe8][0001][cosmos1ahx7f8wyertuus9r20284ej0asrs085case3kn]
 func MakeOrchestratorValidatorAddressKey(orc sdk.AccAddress) []byte {
-	return append([]byte{OrchestratorValidatorAddressKey}, orc.Bytes()...)
+	return bytes.Join([][]byte{{OrchestratorToValidatorAddressKey}, orc.Bytes()}, []byte{})
 }
 
-// MakeValidatorEthereumAddressKey returns the following key format
+// MakeValidatorEVMAddressKeyForValidator returns the following key format
 // prefix              cosmos-validator
-// [0x0][cosmosvaloper1ahx7f8wyertuus9r20284ej0asrs085case3kn]
-func MakeValidatorEthereumAddressKey(validator sdk.ValAddress) []byte {
-	return append([]byte{ValidatorEthereumAddressKey}, validator.Bytes()...)
+// [0x0][0001][cosmosvaloper1ahx7f8wyertuus9r20284ej0asrs085case3kn]
+func MakeValidatorEVMAddressKeyForValidator(validator sdk.ValAddress) []byte {
+	return bytes.Join([][]byte{{ValidatorToEVMAddressKey}, validator.Bytes()}, []byte{})
 }
 
-// MakeEthereumOrchestratorAddressKey returns the following key format
+// MakeEVMOrchestratorAddressKey returns the following key format
 // prefix              cosmos-validator
-// [0x0][cosmosvaloper1ahx7f8wyertuus9r20284ej0asrs085case3kn]
-func MakeEthereumOrchestratorAddressKey(eth common.Address) []byte {
-	return append([]byte{EthereumOrchestratorAddressKey}, eth.Bytes()...)
+// [0x0][0001][cosmosvaloper1ahx7f8wyertuus9r20284ej0asrs085case3kn]
+func MakeEVMOrchestratorAddressKey(eth common.Address) []byte {
+	return bytes.Join([][]byte{{EVMToOrchestratorAddressKey}, eth.Bytes()}, []byte{})
 }
 
 /////////////////////////
 // Etheruem Signatures //
 /////////////////////////
 
-// MakeEthereumSignatureKey returns the following key format
+// MakeEVMSignatureKeyForValidator returns the following key format
 // prefix   nonce                    validator-address
 // [0x0][0 0 0 0 0 0 0 1][cosmos1ahx7f8wyertuus9r20284ej0asrs085case3kn]
-func MakeEthereumSignatureKey(storeIndex []byte, validator sdk.ValAddress) []byte {
-	return bytes.Join([][]byte{{EthereumSignatureKey}, storeIndex, validator.Bytes()}, []byte{})
+func MakeEVMSignatureKeyForValidator(chainID uint32, storeIndex []byte, validator sdk.ValAddress) []byte {
```

Comment:
This does change the input, so likely the comment above would no longer be valid.

Reply:
done